src/core/util/sys_vars.cpp

```diff
@@ -1823,6 +1825,13 @@ void mce_sys_var::get_env_params()
         }
         multilock = (multilock_t)temp;
     }
+
+    if ((env_ptr = getenv(SYS_VAR_XLIO_THREADS))) {
+        xlio_threads = option_size::from_str(env_ptr) ?: 1;
+        if (xlio_threads > std::thread::hardware_concurrency()) {
+            xlio_threads = std::thread::hardware_concurrency();
+        }
+    }
```

Comment:
According https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency `td::thread::hardware_concurrency` can be 0

Should not be reported info message about setting not requested value as done in many cases for example https://github.com/Mellanox/libxlio/blob/1a600a354d1495f0fcd61195d7f6af6d5990a817/src/core/util/sys_vars.cpp#L1283